code/mob/living/carbon/human/procs/Life.dm

```diff
@@ -310,18 +310,43 @@
 
 				if (!myID)
 					myID = wear_id
-				if (myID && (access_carrypermit in myID.access))
+				if (myID && (access_carrypermit in myID.access) && (access_contrabandpermit in myID.access)) // has all permissions for contraband, don't check
 					myID = null
 				else
 					var/contrabandLevel = 0
-					if (l_hand)
-						contrabandLevel += l_hand.contraband
-					if (!contrabandLevel && r_hand)
-						contrabandLevel += r_hand.contraband
-					if (!contrabandLevel && belt)
-						contrabandLevel += belt.contraband
-					if (!contrabandLevel && wear_suit)
-						contrabandLevel += wear_suit.contraband
+					if (myID)
+						var/has_carry_permit = (access_carrypermit in myID.access)
+						var/has_contraband_permit = (access_contrabandpermit in myID.access)
+						if (l_hand)
+							if (istype(l_hand, /obj/item/gun/))
+								if(!has_carry_permit)
+									contrabandLevel += l_hand.contraband
+							else
+								if(!has_contraband_permit)
+									contrabandLevel += l_hand.contraband
+
+						if (!contrabandLevel && r_hand)
+							if (istype(r_hand, /obj/item/gun/))
+								if(!has_carry_permit)
+									contrabandLevel += r_hand.contraband
+							else
+								if(!has_contraband_permit)
+									contrabandLevel += r_hand.contraband
+
+						if (!contrabandLevel && !has_contraband_permit)
+							if (belt)
+								contrabandLevel += belt.contraband
```

Comment:
Shouldn't a firearms permit allow you to carry a gun on your belt safely?

Reply:
Applies to all other contraband checks

Reply:
I only accounted for the wrestling belt as a belt contraband item, forgot about weapons being possible to carry there. Will correct.